editor/gui/scene_tree_editor.cpp

```diff
@@ -925,6 +925,26 @@ void SceneTreeEditor::_notification(int p_what) {
 
 			_update_tree();
 		} break;
+
+		case NOTIFICATION_VISIBILITY_CHANGED: {
+			if (is_visible()) {
+				TreeItem *item = nullptr;
+				if (selected) {
+					// Scroll to selected node.
+					item = _find(tree->get_root(), selected->get_path());
+				} else if (marked.size() == 1) {
+					// Scroll to a single marked node.
+					Node *marked_node = *marked.begin();
+					if (marked_node) {
+						item = _find(tree->get_root(), marked_node->get_path());
+					}
+				}
+
+				if (item) {
+					tree->scroll_to_item(item, true);
```

Comment:
This doesn't seem to work reliably. The item is not always centered when you open the dialog multiple times for different nodes. Might require deferring or waiting until the dialog content is drawn.

Reply:
Oh you're right! I was able to reproduce it when picking Root Node for AnimationPlayer.
Adding a `call_deferred` was indeed enough to make it consistently work.